src/HexagonOptimize.cpp

```diff
@@ -1804,6 +1804,88 @@ class VtmpyGenerator : public IRMutator {
         IRMutator::visit(op);
     }
 };
+
+// Convert some expressions to an equivalent form which could get better
+// optimized in later stages for hexagon
+class RearrangeExpressions : public IRMutator {
+private:
+    using IRMutator::visit;
+
+    void visit(const Mul *op) {
+        if (op->type.is_vector()) {
+            if (op->a.as<Broadcast>()) {
+                // Ensures broadcasts always occurs as op1 not op0
```

Comment:
Sanity check to avoid infinite recursion:

    internal_assert(!op->b.as<Broadcast>());